lib/matplotlib/axes/_axes.py

```diff
@@ -6758,6 +6758,8 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
             labels = [None]
         elif isinstance(label, str):
             labels = [label]
+        elif not np.iterable(label):
+            labels = [str(label)]
```

Comment:
I think you can combine this case with the one above it

Reply:
Unfortunately strings are iterable, so I don't think I can...

```
In [1]: boo = 'who'

In [2]: import numpy as np

In [3]: np.iterable(boo)
Out[3]: True
```

Reply:
This is exactly the code required to make int labels work.

Reply:
My mistake, I thought `np.iterable` ignored strings. `np.ndim(boo) == 0` would work, but is perhaps slower.

Reply:
`np.ndim` looks useful.  I quite readily get confused with the best way to test if something is a list, ndarray, or singleton, and our codebase uses a few incantations.  I would be fine w/ changing this to something that is an agreed-upon standard way to check.  This logic only gets called when a hist is added to the axes, so I don't think we need to worry overly much about performance.